STX.REST.RESTFulSense.Clients/Services/Foundations/ErrorMappers/ErrorMapperService.cs

```diff
@@ -0,0 +1,28 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Threading.Tasks;
+using STX.REST.RESTFulSense.Clients.Brokers.Errors;
+using STX.REST.RESTFulSense.Clients.Models.Errors;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.ErrorMappers
+{
+    internal partial class ErrorMapperService : IErrorMapperService
+    {
+        private readonly IErrorBroker errorBroker;
+
+        public ErrorMapperService(IErrorBroker errorBroker) =>
+            this.errorBroker = errorBroker;
+
+        public ValueTask<StatusDetail> RetrieveStatusDetailByStatusCodeAsync(int statusCode) =>
+            TryCatch(() =>
+            {
+                ValidateStatusCode(statusCode);
+                IQueryable<StatusDetail> statusDetails = errorBroker.SelectAllStatusDetails();
+
+                return new ValueTask<StatusDetail>(statusDetails.FirstOrDefault(statusDetail => statusDetail.Code == statusCode));
```

Comment:
```suggestion
                ValidateStatusCode(statusCode);
                
                StatusDetail statusDetail = errorBroker.SelectAllStatusDetails()
                    .FirstOrDefault(statusDetail => statusDetail.Code == statusCode);

                return await ValueTask.FromResult(statusDetails);
```suggestion
                ValidateStatusCode(statusCode);
                IQueryable<StatusDetail> statusDetails = errorBroker.SelectAllStatusDetails();

                return new ValueTask<StatusDetail>(statusDetails.FirstOrDefault(statusDetail => statusDetail.Code == statusCode));
```
```

Reply:
Treated